integrations/opennms-wsman-asset-provisioning-adapter/src/main/java/org/opennms/netmgt/config/WsManAssetAdapterConfigManager.java

```diff
@@ -0,0 +1,160 @@
+     * simultaneously.
+     */
+    protected void reloadXML(final long lastModified, final InputStream stream) throws IOException {
+        getWriteLock().lock();
+        try(final Reader reader = new InputStreamReader(stream)) {
+            m_config = JaxbUtils.unmarshal(WsManAssetAdapterConfiguration.class, reader);
+            m_lastModified = lastModified;
+        } finally {
+            getWriteLock().unlock();
+        }
+    }
+
+    /**
+     * <p>Update</p>
+     *
+     * @throws java.io.IOException if any.
+     */
+    @Override
+    public void update() throws IOException {
+        getWriteLock().lock();
+        try {
+            final File cfgFile = ConfigFileConstants.getFile(ConfigFileConstants.WSMAN_ASSET_ADAPTER_CONFIG_FILE_NAME);
+            final long lastModified = cfgFile.lastModified();
+            if (lastModified > m_lastModified) {
+                LOG.debug("init: config file path: {}", cfgFile.getPath());
+                reloadXML(lastModified, new FileInputStream(cfgFile));
+                LOG.debug("init: finished loading config file: {}", cfgFile.getPath());
+            }
+        } finally {
+            getWriteLock().unlock();
+        }
+    }
+
+    /**
+     * Return the configuration object.
+     *
+     * @return a {@link org.opennms.netmgt.config.snmpAsset.adapter.WsManAssetAdapterConfiguration} object.
+     */
+    protected WsManAssetAdapterConfiguration getConfiguration() {
+        getReadLock().lock();
+        try {
+            return m_config;
+        } finally {
+            getReadLock().unlock();
+        }
+    }
+
+    /**
+     * Returns all {@link AssetField} objects that are in packages that match the specified
+     * Vendor precisely. This relies on the WsManDetector issuing an Identify and populating
+     * the Vendor and Product fields.
+     */
+    @Override
+    public AssetField[] getAssetFieldsForAddress(final InetAddress address, final String vendor) {
+        getReadLock().lock();
+        
+        try {
+            if (vendor == null) {
+                LOG.debug("getAssetFieldsForAddress: WSMAN vendor field was null for address {}, returning empty list", InetAddressUtils.str(address));
+                return new AssetField[0];
+            }
+    
+            final List<AssetField> retval = new ArrayList<>();
+            for (final org.opennms.netmgt.config.wsmanAsset.adapter.Package pkg : m_config.getPackages()) {
+                final String pkgVendor = pkg.getVendor();
+                if (pkgVendor != null) {
+                    if (pkgVendor.equals(vendor)) {
+                        retval.addAll(pkg.getAssetFields());
+                    }
+                }
+            }
+            if (retval.IsEmpty()) {
+                LOG.debug("getAssetFieldsForAddress: Zero AssetField matches returned for {} with vendor: {}", InetAddressUtils.str(address), vendor);
+            }
+            return retval.toArray();
```

Comment:
Oops, I mistakenly asked you to change this in earlier review. Should have been left alone. Suggestion below reverts it back.

```suggestion
            return retval.toArray(new AssetField[0]);
```